jcl/src/java.base/share/classes/jdk/internal/vm/Continuation.java

```diff
@@ -228,17 +229,19 @@ public static boolean yield(ContinuationScope scope) {
 			} else {
 				throw new AssertionError("Unknown pinned error code: " + rcPinned);
 			}
-			throw new IllegalStateException("Continuation is pinned: " + reason);
+			cont.onPinned(reason);
+		} else {
+			yieldImpl();
+			cont.onContinue();
 		}
-		return yieldImpl();
+		return (rcPinned == 0);
 	}
 
 	protected void onPinned(Pinned reason) {
-		throw new UnsupportedOperationException();
+		throw new IllegalStateException("Continuation is pinned: " + reason);
 	}
 
 	protected void onContinue() {
-		throw new UnsupportedOperationException();
 	}
```

Comment:
Can this method be removed since it is empty?

Reply:
yes, but as it is protected, any JCL code that wish to extend Continuation class may override this method to perform post yield operation (no JCL code do so at the moment)

Reply:
can we add it when it is required? currently, it is an unneeded call, which regresses perf.

Reply:
Since we would not be able to know when JCL code decide to make use of this, I prefer to keep it as is to guarantee future correctness. I am not sure if there is any way of knowing when to add this back other than triaging failed testcases which depends on this.